child_compassion/model/child_compassion.py

```diff
@@ -357,6 +359,32 @@ def _create_empty_case_study(self, cr, uid, child, context=None):
             return child_prop_obj.create(cr, uid, vals, context)
         return True
 
+    def update_delegate(self, cr, uid, context=None):
+        obj_children = self.pool.get('compassion.child')
+        obj_undelegate_wizard = self.pool.get('undelegate.child.wizard')
+
+        child_ids = obj_children.search(cr, uid, [], context=context)
+        child_ids_to_delegate = []
+        child_ids_to_undelegate = []
+
+        for child in obj_children.browse(cr, uid, child_ids, context=context):
+            if child.date_delegation:
+                if datetime.strptime(child.date_delegation, DF) \
+                   <= datetime.today():

```

Comment:
you can add the condition that child.state must be in a valid state (same as the one tested in delegate_wizard).